src/components/menu/menu-popover.tsx

```diff
@@ -10,10 +10,10 @@ export const MenuPopover: FC = ({ children }) => {
   useClickOutside(popoverRef, closeMenu)
 
   if (isOpen) {
-    const onClick = (e) => {
-      // close menu if button or any other element inside e.g icon is clicked
-      const button = e.target.closest('button')
-      if (button) closeMenu()
+    const onClick = (event) => {
+      // close menu if clicked target has parent with below selector
+      const menuItem = event.target.closest('[data-close-menu="true"]')
+      if (menuItem) closeMenu()
```

Comment:
> suggestion: In future, when we discover an issue that has no obvious solution, it's really important that we discuss the tradeoffs involved in the different solutions we can think of before we go and submit code for review.

yes, I'll note this, thanks 👍 

>question: Can you please explain some of your reasoning behind the use of data-close-menu in the manner shown here and contrast it with other possible solutions?

certainly, so as we've discussed previously I think we need to give an options for menu item when it clicked whether it will close the menu or not. and this current pr the default behaviour is to close the menu by default when item clicked, and if user want to prevent that user can set ```<Menu.Item closeMenu={false}>...</Menu.Item>```.

the other alternatives I can think of is:
1. to put the `closeMenu` props in the menu group or the parent list instead so the all the grouped items will have the same behaviour, I think its better but less flexible.
2. to automatically decide the behaviour based on element used in the Items (e.g close the menu for buttons, div with role=button .etc). as currently I use div this is not applicable.
3. strictly define, e.g for simple/basic menu it will close but not for checkbox/radio menu. I'm not sure with this because I think consumer could want checbox to close immediately .etc

or is there other approach that you prefer ? 🤔

Reply:
Thanks for the explanation 🙏 

> is there other approach that you prefer?

Personally, I think leveraging `event.preventDefault()` is the cheapest, most flexible approach we could take for this right now.

1. It doesn't commit us to any specific component interface (like `closeMenu` currently does);
2. I suspect we can safely rely on `:where([role="menuitem"], [role="menuitemradio"], [role="menuitemcheckbox"]` as the selector we use with `event.target.closest`;
3. It doesn't risk confusing CSS-only consumers if they see `data-close-menu` in the docs and think that it works out-of-the-box for them; and,
4. It feels (subjectively) more natural to me than using a data attribute to affect whether the default click behaviour occurs or not. That is, `event.preventDefault()` seem specifically geared towards this kind of thing (to me at least).

That said, what you've done here is still a simple solution and while it commits us to the `closeMenu` prop, it's only one prop, and it wouldn't be hard to migrate away from in future if we had to. So I think the main concern (which may not be important enough to warrant changing what you've done) is point (3): possible confusion for CSS only consumers 🤷‍♂️ 

What do you think?

Reply:
**aside:** Also, when we get access to the `popover` attribute in React 19, we'd be able to support a full HTML-only solution (for button-based menu items, at least) using something like:

```html
<div id="123" role="menu" popover>
  <button role="menuitem" popovertarget="123" popovertargetaction="hide">Item 1</button>
  <button role="menuitem">Item 2</button>
</div>
```

In this case, Item 1 would close the menu popover when clicked, but Item 2 would not.

Reply:
> Personally, I think leveraging event.preventDefault() is the cheapest, most flexible approach we could take for this right now.

I think there is a misunderstanding here, because after I tried `preventDefault()` will have no effect since the onclick handler assigned to the child will run first

![image](https://github.com/user-attachments/assets/e4886444-2ff4-4c85-8c4e-b37c4c303555)

and that is what I expect because my Idea is actually to expand the `onClick` handler of the `Menu.Item` so we don't have to handle in each of them

> I suspect we can safely rely on :where([role="menuitem"], [role="menuitemradio"], [role="menuitemcheckbox"] as the selector we use with event.target.closest;

as currently `data-close-menu` is always on `Menu.Item` I think its enough already for the selector

> It doesn't risk confusing CSS-only consumers if they see data-close-menu in the docs and think that it works out-of-the-box for them; and,

yes, I think I need to add some notes in the docs

> What do you think?

because the docs for css-only user don't show the `data-close-menu` I think it is safe to use my current approach, I'm not sure if there is better aproach than using popover api like what you've mention before 🤔

Reply:
> I tried preventDefault() will have no effect since the onclick handler assigned to the child will run first

I think the misunderstanding may be that it's the child click handler that would call `event.preventDefault()`, and the parent click handler would check if `event.defaultPrevented` is true or not before closing the menu.

All good though. Like I said, you can make the call on this one and, hopefully sometime soon, we'll get all the nice new native functionality from the popover attribute 😄

Reply:
great, thanks for the comment so far it really give me more insights😄 

I will make the amendments as per the discussion previously 👍